Homework_36.ts

```diff
@@ -0,0 +1,169 @@
+					actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+					plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+					country: "United Kingdom, United States",
+					poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+					imdbRating: 7.6,
+					imdbVotes: 684604,
+					boxOffice: "$318,087,620",
+					production: "1492 Pictures, Heyday Films, Warner Brothers",
+				}
+    ]
+
+    //1. Собрать в массив все жанры фильмов (без повторения)
+
+    const getGenre = (array: Movies[]) => {
+        const genre = array.map(movie=> movie.genre)
+        const arrayGenres = []
+        return Array.from(new Set(arrayGenres.concat(...genre)))
+    }
+    console.log('All genres', getGenre(movies))
+
+    // 2. Собрать в массив всех актеров всех фильмов (без повторения)
+        
+    const getActors = (array: Movies[]) => {
+        const actors = array.map(movie=> movie.actors)
+        const arrayActors = []
+        return Array.from(new Set(arrayActors.concat(...actors)))
+    }
+    console.log('All actors', getActors(movies))
+
+    //3. Отсортировать фильмы по рейтингу по убыванию
+
+    const sortRating = (array: Movies[]) =>array.sort((a, b)=> b.imdbRating - a.imdbRating)
+    console.log('sorted by rating', sortRating(movies))
+
+    //4. Создать новый массив, где объекты фильмов будут состоять из следующих полей: id, title, released, plot
+
+    const getNewArrayMovies = (array: Movies[]) =>array.map(movie=>({id: movie.id, title: movie.title, released: movie.released, plot: movie.plot}))
+    console.log(getNewArrayMovies(movies))
+
+    //5. Создать функцию, которая бы принимала массив фильмов и число. А результатом этой функции должен быть отфильтрованный массив,
+    // с фильмами где число равно году выхода фильма.
+
+    const getMovieYear = (array: Movies[], year: number) => array.filter(movie => movie.year === year)
+    console.log(getMovieYear(movies, 2021))
+
+    // 6. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть новый
+    // отфильтрованный массив, с фильмами, где строка входит в название фильма.
+
+    const getMovieWithString = (array: Movies[], string: string) => array.filter(movie => movie.title.includes(string) === true)
+    console.log('6 task', getMovieWithString(movies, 'Harry Potter and'))
+
+    // 7. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть
+    // отфильтрованный массив, с фильмами где строка входит в название фильма или в его сюжет.
+    const getMovieWithStringInTitleOrPlot = (array: Movies[], string: string) => {
+        return Array.from(new Set(array.filter(movie => (movie.title.includes(string) === true) || (movie.plot.includes(string) === true))))
```

Comment:
why do you need `Set` here?

Reply:
Here it is needed so that there are no duplicates of objects in which there are matches in both title and plot

Reply:
it wouldn't work as you expect - as all obj are unique. So {id:1} and {id:1} will both stay in Set
also - it's a bad idea to remove them here. Such a case is a bug in data, you shouldn't fix it on your side.